autoload/ale/scala.vim

```diff
@@ -0,0 +1,23 @@
+" Author: Zoltan Kalmar - https://github.com/kalmiz,
+"         w0rp <devw0rp@gmail.com>
+"         Nils Leuzinger - https://github.com/PawkyPenguin
+" Description: Functions for usage with scalac-like linters
+
+function! ale#scala#GetExecutableForLinter(buffer, linter) abort
+    if index(split(getbufvar(a:buffer, '&filetype'), '\.'), 'sbt') >= 0
+        " Don't check sbt files
+        return ''
+    endif
+
+    return a:linter
+endfunction
+
+function! ale#scala#GetCommandForLinter(buffer, linter) abort
+    let l:executable = ale#scala#GetExecutableForLinter(a:buffer, a:linter)
+
+    if empty(l:executable)
+        return ''
+    endif
+
+    return ale#Escape(l:executable) . ' -Ystop-after:parser %t'
+endfunction
```

Comment:
I'd rather you just copy and paste the `scalac` code into the new `fsc` file, and leave the old code alone. They are similar at the moment, but both using the same functions like this will make it harder to modify one without breaking the other one. `scalac` and `fsc` probably won't support _exactly_ the same options at some point, say if one program is updated before another.